src/transformers/image_utils.py

```diff
@@ -202,7 +203,12 @@ def infer_channel_dimension_format(
     else:
         raise ValueError(f"Unsupported number of image dimensions: {image.ndim}")
 
-    if image.shape[first_dim] in num_channels:
+    if image.shape[first_dim] in num_channels and image.shape[last_dim] in num_channels:
+        warnings.warn(
+            f"The channel dimension is ambiguous. Got image shape {image.shape}. Assuming channels are the first dimension. Please specify with input_data_format if this is incorrect."
+        )
```

Comment:
* The function shouldn't have any knowledge about what is calling it. Just flagging this assumption is happening should be enough
* It's better to use `logger.warning` as we can control this through logger reporting levels

```suggestion
        logger.warning(
            f"The channel dimension is ambiguous. Got image shape {image.shape}. Assuming channels are the first dimension."
        )
```